code/Model/EditableFormField/EditableFileField.php

```diff
@@ -221,14 +238,22 @@ public function getFormField()
         }
 
         $folder = $this->Folder();
+        
+        if ($protectFolder && $folder->ID === 0) {
+            Versioned::set_stage($this->config()->get('file_upload_stage'));
+
+            $folderName = sprintf('%s_%s', $this->ParentID, $this->ID);
```

Comment:
Noticed this is separated with a slash rather than a forward slash, means there will be one less level of folder nested, though that's probably good since there should be very few of these sorts of folders ever created